### PR TITLE
Domains: Remove Change Site Address on paid domains

### PR DESCRIPTION
**Pull request by lupus2k**
Related to https://github.com/Automattic/nomado-issues/issues/456#top

## Proposed Changes

We want to not show the `Change site name` option for non-managed WPCOM subdomains.

| .com | .wordpress.com |
| --- | --- |
| image | image |

## Testing Instructions

* Go to `/domains/manage/:site-slug` with a custom domain
* The custom domain should not show the option in the 3 dots menu.
* The wpcom domain should show the option.

### CODE REVIEW

File: packages/domains-table/src/utils/is-free-url-domain-name.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (paulopmt1) [inline packages/domains-table/src/utils/is-free-url-domain-name.ts]: Is there any limitation for us [to reuse that original code](https://github.com/Automattic/wp-calypso/blob/19451117505be00b465f99767aa8b633fa0c6b28/client/lib/domains/utils/is-free-url-domain-name.ts)? If so, can we refactor to use one single source?

**Review by paulopmt1 — COMMENTED** [review]
It's working as expected. Curious about that duplicated code.

File: packages/domains-table/src/utils/is-free-url-domain-name.ts (hunk 1 above)
Comment (lupus2k) [reply packages/domains-table/src/utils/is-free-url-domain-name.ts]: I had this lint error, so I think is better to keep it inside the domains-table package
`error  'calypso/lib/domains/utils' import is restricted from being used by a pattern  no-restricted-imports`

Comment (lupus2k) [reply packages/domains-table/src/utils/is-free-url-domain-name.ts]: Do you think ignoring the lint could be a better approach?

Comment (paulopmt1) [reply packages/domains-table/src/utils/is-free-url-domain-name.ts]: Oh, I see we're editing package files...

Maybe we could do the other way around, by [removing that original file](https://github.com/Automattic/wp-calypso/blob/HEAD/client/lib/domains/utils/is-free-url-domain-name.ts#L0-L1) and changing the imports [like this one ](https://github.com/Automattic/wp-calypso/blob/fix/hide-change-site-name-on-non-free-domains/client/my-sites/domains/domain-management/controller.jsx#L3) to `import { isFreeUrlDomainName } from '@automattic/domains-table/src/utils/is-free-url-domain-name';`?

My main concern is that it seems we need to maintain those domains updated as we add new free sub-domains and having one place only would help with that.

Comment (paulopmt1) [reply packages/domains-table/src/utils/is-free-url-domain-name.ts]: [Added that change here](https://github.com/Automattic/wp-calypso/pull/83553/commits/33c12a51e2431469c5e27a96c99329a0cec5f382). @delputnam could you please double check everything is working as expected?

**Review by paulopmt1 — APPROVED** [review]
Working as expected, thanks @lupus2k!

I also added a fix for the unique key console error:

image

File: packages/domains-table/src/utils/is-free-url-domain-name.ts (hunk 1 above)
Comment (lupus2k) [reply packages/domains-table/src/utils/is-free-url-domain-name.ts]: Thanks for your update @paulopmt1, it looks great!

**Comment by lupus2k** [comment]
> I also added a fix for the unique key console error:

That's great, thanks @paulopmt1

**Review by DustyReagan — APPROVED** [review]
Testing well for me! Nice fix!

**Pull request merged**